diff --git a/aidl/Android.bp b/aidl/Android.bp
index 61c21ed..af1ed74 100644
--- a/aidl/Android.bp
+++ b/aidl/Android.bp
@@ -37,7 +37,6 @@ cc_library_shared {
     name: "vendor.qti.hardware.vibrator.impl",
     defaults: [
         "vibrator_defaults",
-        "qti_vibrator_hal_defaults",
     ],
     vendor: true,
     srcs: [
@@ -48,6 +47,7 @@ cc_library_shared {
         "libcutils",
         "libutils",
         "liblog",
+        "libqtivibratoreffect",
         "libqtivibratoreffectoffload",
         "libbinder_ndk",
     ],
