diff --git a/Android.mk b/Android.mk
index 78b9430..ff9296c 100755
--- a/Android.mk
+++ b/Android.mk
@@ -334,14 +334,14 @@ ifeq ($(TW_INCLUDE_CRYPTO), true)
     TW_INCLUDE_CRYPTO_FBE := true
     LOCAL_CFLAGS += -DTW_INCLUDE_FBE
     LOCAL_SHARED_LIBRARIES += android.frameworks.stats@1.0 android.hardware.authsecret@1.0 \
-	android.security.authorization-ndk_platform \
+	android.security.authorization-ndk \
         android.hardware.oemlock@1.0 libf2fs_sparseblock \
         libandroidicu.recovery \
         android.hardware.gatekeeper@1.0 \
         android.hardware.weaver@1.0 \
         android.frameworks.stats@1.0 \
-        android.security.maintenance-ndk_platform \
-        android.system.keystore2-V1-ndk_platform \
+        android.security.maintenance-ndk \
+        android.system.keystore2-V1-ndk \
         libkeyutils \
         liblog \
         libsqlite.recovery \
diff --git a/libtar/Android.mk b/libtar/Android.mk
index c3bb691..008125a 100755
--- a/libtar/Android.mk
+++ b/libtar/Android.mk
@@ -23,10 +23,10 @@ ifeq ($(TW_INCLUDE_CRYPTO_FBE), true)
         android.hardware.keymaster@4.0 \
         android.hardware.keymaster@4.1 \
         android.hardware.weaver@1.0 \
-        android.security.apc-ndk_platform \
-        android.system.keystore2-V1-ndk_platform \
-        android.security.authorization-ndk_platform \
-        android.security.maintenance-ndk_platform \
+        android.security.apc-ndk \
+        android.system.keystore2-V1-ndk \
+        android.security.authorization-ndk \
+        android.security.maintenance-ndk \
         libselinux \
         libbinder_ndk \
         libext4_utils \
